src/toggle-switch/toggle-switch.scss

```diff
@@ -2,194 +2,209 @@
 // See LICENSE.md in the project root for license terms and full copyright notice.
 @import '../style/index';
 
-@mixin iui-toggle-switch {
+@mixin iui-toggle-switch-wrapper {
   @include iui-reset;
   display: flex;
   align-items: center;
+  gap: $iui-s;
+  align-items: center;
   font-size: $iui-font-size;
   width: fit-content;
-  padding: ($iui-baseline/2) 0;
   user-select: none;
   cursor: pointer;
-  position: relative;
+  isolation: isolate;
   @include themed {
     color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-2));
   }
 
+  // Disabled
+  &.iui-disabled {
+    cursor: not-allowed;
+    @include themed {
+      color: t(iui-text-color-muted);
+    }
+  }
+
+  // IE solution
+  // For labels on the left
+  > .iui-label {
+    margin-right: $iui-s;
+    @supports (gap: $iui-s) {
+      margin-right: 0;
+    }
+  }
+
+  // For labels on the right
+  .iui-toggle-switch ~ .iui-label {
+    margin-left: $iui-s;
+    margin-right: 0;
+    @supports (gap: $iui-s) {
+      margin-left: 0;
+    }
+  }
+}
+
+@mixin iui-toggle-switch {
+  --_iui-toggle-switch-handle-size: #{$iui-icons-default};
+  --_iui-toggle-switch-handle-offset: #{$iui-component-offset};
+  --_iui-toggle-switch-border-thickness: 1px;
+
+  display: grid;
+  grid-template: 'toggle';
+  align-items: center;
+
   > input {
-    width: 0;
-    height: 0;
-    appearance: none;
-    opacity: 0;
-    position: absolute;
+    @include iui-reset;
+    @include iui-focus($thickness: 1px, $offset: 1px);
+    grid-area: toggle;
+    display: flex;
+    position: relative;
+    cursor: pointer;
+    -webkit-tap-highlight-color: transparent;
+    touch-action: pan-y;
```

Comment:
What it does?

Reply:
This tells the browser that it should only handle the y-axis for drag events. It's useful when you have custom drag logic for touchscreen devices (e.g. our Slider component) but it's not helping in this component

Reply:
The idea was taken from this article when I was searching for a centering solution. https://web.dev/building-a-switch-component/#touch-action

Reply:
Removing it because I just want to get this PR approved, can add it later if wanted.